Controller/Transaction/CommitWebpayM22.php

```diff
@@ -0,0 +1,225 @@
+                                \Magento\Checkout\Model\Session $checkoutSession,
+                                \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
+                                \Magento\Framework\Controller\Result\RawFactory $resultRawFactory,
+                                \Transbank\Webpay\Model\Config\ConfigProvider $configProvider) {
+
+        parent::__construct($context);
+
+        $this->cart = $cart;
+        $this->checkoutSession = $checkoutSession;
+        $this->resultJsonFactory = $resultJsonFactory;
+        $this->resultRawFactory = $resultRawFactory;
+        $this->messageManager = $context->getMessageManager();
+        $this->configProvider = $configProvider;
+        $this->log = new LogHandler();
+    }
+
+    /**
+     * @Override
+     */
+    public function execute() {
+
+        $config = $this->configProvider->getPluginConfig();
+        $orderStatusCanceled = $config['error_pay'];
+
+        try {
+
+            $order = $this->getOrder();
+
+            $tokenWs = isset($_POST['token_ws']) ? $_POST['token_ws'] : null;
+
+            if($tokenWs != $this->checkoutSession->getTokenWs()) {
+                throw new \Exception('Token inválido');
+            }
+
+            $paymentOk = $this->checkoutSession->getPaymentOk();
+
+            if ($paymentOk  == 'WAITING') {
+
+                $transbankSdkWebpay = new TransbankSdkWebpay($config);
+                $result = $transbankSdkWebpay->commitTransaction($tokenWs);
+
+                $this->checkoutSession->setResultWebpay($result);
+
+                if (isset($result->buyOrder) && isset($result->detailOutput) && $result->detailOutput->responseCode == 0) {
+
+                    $this->checkoutSession->setPaymentOk('SUCCESS');
+
+                    $authorizationCode = $result->detailOutput->authorizationCode;
+                    $payment = $order->getPayment();
+                    $payment->setLastTransId($authorizationCode);
+                    $payment->setTransactionId($authorizationCode);
+                    $payment->setAdditionalInformation([\Magento\Sales\Model\Order\Payment\Transaction::RAW_DETAILS => (array)$result]);
+
+                    $orderStatus = $config['sucefully_pay'];
+                    $order->setState($orderStatus)->setStatus($orderStatus);
+                    $order->addStatusToHistory($order->getStatus(), json_encode($result));
+                    $order->save();
+
+                    $this->checkoutSession->getQuote()->setIsActive(false)->save();
+
+                    return $this->toRedirect($result->urlRedirection, array('token_ws' => $tokenWs));
+
+                } else {
+
+                    $this->checkoutSession->setPaymentOk('FAIL');
+
+                    $order->setState($orderStatusCanceled)->setStatus($orderStatusCanceled);
+                    $order->addStatusToHistory($order->getStatus(), json_encode($result));
+                    $order->save();
```

Comment:
why not make these lines be a method?

Reply:
this can be a good improvement to other PR :).